plugins/metalink/utils.c

```diff
@@ -471,8 +459,12 @@ TDNFXmlParseData(
 
     if(!elementInfo || !elementInfo->ml_ctx || IsNullOrEmptyString(elementInfo->filename) || (elementInfo->dwError != 0))
     {
-        elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
-        BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+        if (elementInfo) {
+            elementInfo->dwError = ERROR_TDNF_INVALID_PARAMETER;
+            BAIL_ON_TDNF_ERROR(elementInfo->dwError);
+        }
+        pr_err("%s:%d elementInfo is NULL\n", __func__, __LINE__);
+        return;
```

Comment:
Good catching the `NULL` pointer check, but I think there should be a `BAIL_ON_TDNF_ERROR` here.